### PR TITLE
Add default content-length header in HTTP/2

### LINKED ISSUE #208: Add default content-length header to HTTP/2 requests

### PR DESCRIPTION
**Pull request by whatyouhide**
Closes #208.

### CODE REVIEW

File: lib/mint/http.ex [hunk 1]
```diff
@@ -440,6 +440,14 @@ defmodule Mint.HTTP do
   For a quick discussion on HTTP/2 streams and requests, see the `Mint.HTTP2` module and
   `Mint.HTTP2.request/6`.
 
+  ## The `content-length` header
+
+  If you don't set the `content-length` header and you send a body with the request (that
+  is, not `nil` and not `:stream`), then Mint will add a default `content-length` header
+  to your request. If you're using HTTP/2 and streaming the request, provide the
```
Comment (ericmj) [inline lib/mint/http.ex]: It is optional so I suggest this change:

```suggestion
  to your request. If you're using HTTP/2 and streaming the request, you may provide the
```

File: lib/mint/http.ex [hunk 2]
```diff
@@ -440,6 +440,14 @@ defmodule Mint.HTTP do
...
+  `content-length` header yourself. If you're using HTTP/1, Mint might do chunked
```
Comment (ericmj) [inline lib/mint/http.ex]:
```suggestion
  `content-length` header yourself. If you're using HTTP/1, Mint will do chunked
```

File: lib/mint/http.ex [hunk 3]
```diff
@@ -440,6 +440,14 @@ defmodule Mint.HTTP do
...
+  transfer-encoding by default (see `Mint.HTTP1.request/6`).
```
Comment (ericmj) [inline lib/mint/http.ex]:
```suggestion
  transfer-encoding when a content-length is not provided (see `Mint.HTTP1.request/6`).
```

File: lib/mint/core/util.ex [hunk 4]
```diff
@@ -79,6 +79,14 @@ defmodule Mint.Core.Util do
     end
   end
 
+  def put_new_header_lazy(headers, name, fun) do
+    cond do
+      List.keymember?(headers, name, 0) -> headers
+      value = fun.() -> [{name, value} | headers]
```
Comment (ericmj) [inline lib/mint/core/util.ex]: I can't find an example when this would return `nil` so it seems the `true` clause would never hit?

Comment (whatyouhide) [reply lib/mint/core/util.ex]: Not really any value in handling `nil` except for consistency with the `put_new_header/3` function. I'll remove the handling of `nil`.

**Comment by whatyouhide** [comment]
@ericmj done, ready for another review.

**Pull request merged**